cores/esp8266/core_esp8266_postmortem.h

```diff
@@ -0,0 +1,108 @@
+/*
+ Part of the postmortem.c - output of debug info on sketch crash
+ Copyright (c) 2015 Ivan Grokhotkov. All rights reserved.
+ This file is part of the esp8266 core for Arduino environment.
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <rst_info.h> // NONOS SDK `struct rst_info`, ref. user_interafce.h
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// using numbers different from "REASON_" in user_interface.h (=0..6)
+enum user_rst_reason
+{
+    REASON_USER_ABORT = 249,
+    REASON_USER_ASSERT = 250,
+    REASON_USER_PANIC = 251,
+    REASON_USER_STACK_OVERFLOW = 252,
+    REASON_USER_STACK_SMASH = 253,
+    REASON_USER_EXCEPTION = 254,
+};
+
+// Following functions could be overridden, simply declare them in your own code with a custom functionality
+// Ensure that overrides are either declared with `extern "C"` *or* located in .c file(s)
+// Comments below describe the default function.
+
+#ifdef __cplusplus
+#define ESP8266_POSTMORTEM_NOEXCEPT noexcept
+#else
+#define ESP8266_POSTMORTEM_NOEXCEPT
+#endif
```

Comment:
General idea here is to restrict crash opportunities in the functions. Even when exceptions are on.